Doc/library/os.rst

```diff
@@ -4173,17 +4137,121 @@ written in Python, such as a mail server's external command delivery program.
+   information.  The arguments to :func:`wait4` are the same as those provided
+   to :func:`waitpid`.
 
    :func:`waitstatus_to_exitcode` can be used to convert the exit status into an
    exitcode.
 
    .. availability:: Unix.
 
 
+.. data:: P_PID
+          P_PGID
+          P_ALL
+          P_PIDFD
+
+   These are the possible values for *idtype* in :func:`waitid`. They affect
+   how *id* is interpreted:
+
+   * With :data:`P_PID`, wait for the child with the PID given as *id*.
+   * With :data:`P_PGID`, wait for any child whose progress group ID is *id*.
+   * With :data:`P_ALL`, wait for any child; *id* is ignored.
+   * With :data:`P_PIDFD`, wait for the child identified by the file descriptor
+     *id* (a process file descriptor created with :func:`pidfd_open`). This is
+     Linux-specific.
+
+   .. availability:: Unix, ``P_PIDFD`` only on Linux 5.4+.
+
+   .. versionadded:: 3.3
+   .. versionadded:: 3.9
+      The ``P_PIDFD`` constant.
+
+
+.. data:: WCONTINUED
+
+   This option for :func:`waitpid`, :func:`wait3`, :func:`wait4`, and
+   :func:`waitid` causes child processes to also be reported if they have been
+   continued from a job control stop since their status was last reported.
+
+   .. availability:: some Unix systems.
+
+
+.. data:: WEXITED
+
+   This option for :func:`waitid` causes child processes that have terminated to
+   be reported.
+
+   The other ``wait*`` function always report children that have terminated,
+   so this option is not available for them.
+
+   .. availability:: Unix.
+
+   .. versionadded:: 3.3
+
+
+.. data:: WSTOPPED
+
+   This option for :func:`waitid` causes child processes that have been stopped
+   by the delivery of a signal to be reported.
+
+   This option is not available for the other ``wait*`` functions.
+
+   .. availability:: Unix.
+
+   .. versionadded:: 3.3
+
+
+.. data:: WUNTRACED
+
+   This option for :func:`waitpid`, :func:`wait3`, and :func:`wait4` causes
+   child processes to also be reported if they have been stopped but their
+   current state has not been reported since they were stopped.
+
+   This option is not available for :func:`waitid`.
+
+   .. availability:: Unix.
+
+
+.. data:: WNOHANG
+
+   The option for :func:`waitpid`, :func:`wait3`, :func:`wait4`, and
+   :func:`waitid` to return immediately if no child process status is available
+   immediately.
+
+   .. availability:: Unix.
+
+
+.. data:: WNOWAIT
+
+   The option for :func:`waitid` to leave the child in a waitable state, so that
+   a later wait call can be used again to retrieve the child status information.
+
+   This option is not available for the other ``wait*`` functions.
+
+   .. availability:: Unix.
+
+
+.. data:: CLD_EXITED
+          CLD_KILLED
+          CLD_DUMPED
+          CLD_TRAPPED
+          CLD_STOPPED
+          CLD_CONTINUED
+
+   These are the possible values for :attr:`si_code` in the result returned by
+   :func:`waitid`.
```

Comment:
Would it be worth including some brief details on each `si_code` value? 

I don't think it would be quite as useful as the `P_*` options, but some are not entirely intuitive. Particularly for `CLD_EXITED`, I would not guess that it's very specifically denoting that the child process called `_exit(2)` without consulting documentation. Others such as `CLD_KILLED`, `CLD_STOPPED`, and `CLD_CONTINUED` are significantly more obvious based on the name, but I think it would still be useful to have a short description with sphinx links to the `signal` module.

Reply:
> I would not guess that it's very specifically denoting that the child process called _exit(2) without consulting documentation

I don't think that's as significant as you think, since `_exit` is also what the C runtime calls when the `main` function finishes normally. So it really is just "exited normally".